Include/object.h

```diff
@@ -180,6 +180,12 @@ _Py_ThreadId(void)
     tid = __readfsdword(24);
 #elif defined(_MSC_VER) && defined(_M_ARM64)
     tid = __getReg(18);
+#elif defined(__MINGW32__) && defined(_M_X64)
+    tid = __readgsqword(48);
+#elif defined(__MINGW32__) && defined(_M_IX86)
+    tid = __readfsdword(24);
+#elif defined(__MINGW32__) && defined(_M_ARM64)
+    tid = __getReg(18);
```

Comment:
How about `#if (defined(_MSC_VER) || defined(__MINGW32__)) && defined(_M_X64)` since the bodies are the same? (unless this could diverge in the future)

Reply:
I did it this way based on @zooba's preference